Error on start

Hello, i want to test your homebridge. But i have an error when i start the node.

> homebridge@0.0.0 start /home/beber/homebridge
> node app.js

Starting HomeBridge server...

undefined:18
        }
        ^
SyntaxError: Unexpected token }
    at Object.parse (native)
    at Object.<anonymous> (/home/beber/homebridge/app.js:21:19)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:902:3

npm ERR! homebridge@0.0.0 start: `node app.js`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the homebridge@0.0.0 start script.
npm ERR! This is most likely a problem with the homebridge package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node app.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls homebridge
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.19.0-20-generic
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "run" "start"
npm ERR! cwd /home/beber/homebridge
npm ERR! node -v v0.10.25
npm ERR! npm -v 1.4.21
npm ERR! code ELIFECYCLE
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian
 
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs', '/usr/bin/npm', 'run', 'start' ]
2 info using npm@1.4.21
3 info using node@v0.10.25
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info prestart homebridge@0.0.0
6 info start homebridge@0.0.0
7 verbose unsafe-perm in lifecycle true
8 info homebridge@0.0.0 Failed to exec start script
9 error homebridge@0.0.0 start: `node app.js`
9 error Exit status 8
10 error Failed at the homebridge@0.0.0 start script.
10 error This is most likely a problem with the homebridge package,
10 error not with npm itself.
10 error Tell the author that this fails on your system:
10 error     node app.js
10 error You can get their info via:
10 error     npm owner ls homebridge
10 error There is likely additional logging output above.
11 error System Linux 3.19.0-20-generic
12 error command "/usr/bin/nodejs" "/usr/bin/npm" "run" "start"
13 error cwd /home/beber/homebridge
14 error node -v v0.10.25
15 error npm -v 1.4.21
16 error code ELIFECYCLE
17 verbose exit [ 1, true ]

ha you got an idea ?

thank for your solution, it's right.

But now i have a new error :

> homebridge@0.0.0 start /home/beber/homebridge
> node app.js

Starting HomeBridge server...
*** WARNING *** The program 'nodejs' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs>
*** WARNING *** The program 'nodejs' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs&f=DNSServiceRegister>

/home/beber/homebridge/lib/HAP-NodeJS/node_modules/ed25519/node_modules/bindings/bindings.js:83
        throw e
              ^
Error: Symbol ed25519_module not found.
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at bindings (/home/beber/homebridge/lib/HAP-NodeJS/node_modules/ed25519/node_modules/bindings/bindings.js:76:44)
    at Object.<anonymous> (/home/beber/homebridge/lib/HAP-NodeJS/node_modules/ed25519/index.js:1:99)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)

npm ERR! homebridge@0.0.0 start: `node app.js`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the homebridge@0.0.0 start script.
npm ERR! This is most likely a problem with the homebridge package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node app.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls homebridge
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.19.0-20-generic
npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "run" "start"
npm ERR! cwd /home/beber/homebridge
npm ERR! node -v v0.10.25
npm ERR! npm -v 1.4.21
npm ERR! code ELIFECYCLE
npm WARN This failure might be due to the use of legacy binary "node"
npm WARN For further explanations, please read
/usr/share/doc/nodejs/README.Debian
 
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/beber/homebridge/npm-debug.log
npm ERR! not ok code 0

no, i dont' skiped then 

beber@beber-Portable:~/homebridge/lib/HAP-NodeJS$ sudo npm install
npm WARN package.json curve25519@1.0.0 No repository field.
/
> curve25519@1.0.0 install /home/beber/homebridge/lib/HAP-NodeJS/node_modules/curve25519
> node-gyp rebuild

make: Entering directory '/home/beber/homebridge/lib/HAP-NodeJS/node_modules/curve25519/build'
  CXX(target) Release/obj.target/curve/node_curve.o
../node_curve.cc: In function ‘v8::Handle<v8::Value> DoCurve(const v8::Arguments&)’:
../node_curve.cc:28:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
  SOLINK_MODULE(target) Release/obj.target/curve.node
  SOLINK_MODULE(target) Release/obj.target/curve.node: Finished
  COPY Release/curve.node
make: Leaving directory '/home/beber/homebridge/lib/HAP-NodeJS/node_modules/curve25519/build'

> bignum@0.9.2 install /home/beber/homebridge/lib/HAP-NodeJS/node_modules/srp/node_modules/bignum
> node-gyp configure build

make: Entering directory '/home/beber/homebridge/lib/HAP-NodeJS/node_modules/srp/node_modules/bignum/build'
  CXX(target) Release/obj.target/bignum/bignum.o
  SOLINK_MODULE(target) Release/obj.target/bignum.node
  SOLINK_MODULE(target) Release/obj.target/bignum.node: Finished
  COPY Release/bignum.node
make: Leaving directory '/home/beber/homebridge/lib/HAP-NodeJS/node_modules/srp/node_modules/bignum/build'

> ed25519@0.0.2 install /home/beber/homebridge/lib/HAP-NodeJS/node_modules/ed25519
> node-gyp rebuild

make: Entering directory '/home/beber/homebridge/lib/HAP-NodeJS/node_modules/ed25519/build'
  CC(target) Release/obj.target/ed25519/src/ed25519/keypair.o
  CC(target) Release/obj.target/ed25519/src/ed25519/sign.o
  CC(target) Release/obj.target/ed25519/src/ed25519/open.o
  CC(target) Release/obj.target/ed25519/src/ed25519/crypto_verify_32.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_double_scalarmult.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_frombytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_scalarmult_base.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_precomp_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p2_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p2_dbl.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_dbl.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_to_p2.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_to_cached.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p3_tobytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_madd.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_add.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_msub.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_sub.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p1p1_to_p3.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_p1p1_to_p2.o
  CC(target) Release/obj.target/ed25519/src/ed25519/ge_tobytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_0.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_1.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_cmov.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_copy.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_neg.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_add.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_sub.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_mul.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_sq.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_sq2.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_invert.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_tobytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_isnegative.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_isnonzero.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_frombytes.o
  CC(target) Release/obj.target/ed25519/src/ed25519/fe_pow22523.o
  CC(target) Release/obj.target/ed25519/src/ed25519/sc_reduce.o
  CC(target) Release/obj.target/ed25519/src/ed25519/sc_muladd.o
  CXX(target) Release/obj.target/ed25519/src/ed25519.o
  SOLINK_MODULE(target) Release/obj.target/ed25519.node
  SOLINK_MODULE(target) Release/obj.target/ed25519.node: Finished
  COPY Release/ed25519.node
make: Leaving directory '/home/beber/homebridge/lib/HAP-NodeJS/node_modules/ed25519/build'
curve25519@1.0.0 node_modules/curve25519
├── bindings@1.2.1
└── nan@1.8.4

srp@0.2.0 node_modules/srp
└── bignum@0.9.2 (nan@1.6.2)

ed25519@0.0.2 node_modules/ed25519
├── bindings@1.2.1
└── nan@1.8.4
beber@beber-Portable:~/homebridge/lib/HAP-NodeJS$

--- a/app.js
+++ b/app.js
@@ -20,44 +20,15 @@ storage.initSync();
 // Load up the configuration file
 var config = JSON.parse(fs.readFileSync(configPath));
 
+// Just to prevent them getting garbage collected
+var accessories = [];
+
+function startup() {
+    if (config.platforms) loadPlatforms();
+    if (config.accessories) loadAccessories();
+}
+
 function loadAccessories() {
-
-    var accessories = [];
-
-    console.log("Loading " + config.platforms.length + " platforms...");
-    for (var i=0; i<config.platforms.length; i++) {
-
-        var platformConfig = config.platforms[i];
-
-        // Load up the class for this accessory
-        var platformName = platformConfig["platform"]; // like "Wink"
-        var platformModule = require('./platforms/' + platformName + ".js"); // like "./platforms/Wink.js"
-        var platformConstructor = platformModule.platform; // like "WinkPlatform", a JavaScript constructor
-
-        // Create a custom logging function that prepends the platform display name for debugging
-        var name = platformConfig["name"];
-        var log = function(name) { return function(s) { console.log("[" + name + "] " + s); }; }(name);
-
-        log("Initializing " + platformName + " platform...");
-
-        var platform = new platformConstructor(log, platformConfig);
-
-        // query for devices
-        platform.accessories(function(foundAccessories){
-          // loop through accessories adding them to the list and registering them
-          for (var i = 0; i < foundAccessories.length; i++) {
-            accessory = foundAccessories[i]
-            accessories.push(accessory);
-            log("Initializing device with name " + accessory.name + "...")
-            // Extract the raw "services" for this accessory which is a big array of objects describing the various
-            // hooks in and out of HomeKit for the HAP-NodeJS server.
-            var services = accessory.getServices();
-            // Create the HAP server for this accessory
-            createHAPServer(accessory.name, services);
-          }
-          accessories.push.apply(accessories, foundAccessories);
-        })
-    }
 
     // Instantiate all accessories in the config
     console.log("Loading " + config.accessories.length + " accessories...");
@@ -84,6 +55,44 @@ function loadAccessories() {
 
         // Create the HAP server for this accessory
         createHAPServer(name, services);
+    }
+}
+
+function loadPlatforms() {
+
+    console.log("Loading " + config.platforms.length + " platforms...");
+    for (var i=0; i<config.platforms.length; i++) {
+
+        var platformConfig = config.platforms[i];
+
+        // Load up the class for this accessory
+        var platformName = platformConfig["platform"]; // like "Wink"
+        var platformModule = require('./platforms/' + platformName + ".js"); // like "./platforms/Wink.js"
+        var platformConstructor = platformModule.platform; // like "WinkPlatform", a JavaScript constructor
+
+        // Create a custom logging function that prepends the platform display name for debugging
+        var name = platformConfig["name"];
+        var log = function(name) { return function(s) { console.log("[" + name + "] " + s); }; }(name);
+
+        log("Initializing " + platformName + " platform...");
+
+        var platform = new platformConstructor(log, platformConfig);
+
+        // query for devices
+        platform.accessories(function(foundAccessories){
+            // loop through accessories adding them to the list and registering them
+            for (var i = 0; i < foundAccessories.length; i++) {
+                accessory = foundAccessories[i]
+                accessories.push(accessory);
+                log("Initializing device with name " + accessory.name + "...")
+                // Extract the raw "services" for this accessory which is a big array of objects describing the various
+                // hooks in and out of HomeKit for the HAP-NodeJS server.
+                var services = accessory.getServices();
+                // Create the HAP server for this accessory
+                createHAPServer(accessory.name, services);
+            }
+            accessories.push.apply(accessories, foundAccessories);
+        })
     }
 }
 
@@ -175,4 +184,4 @@ function createUsername(str) {
            hash[10] + hash[11];
 }
 
-loadAccessories();
+startup();

